pkg/server/plugin/keymanager/v0_test.go

```diff
@@ -0,0 +1,359 @@
+package keymanager_test
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/server/plugin/keymanager"
+	keymanagerv0 "github.com/spiffe/spire/proto/spire/plugin/server/keymanager/v0"
+	"github.com/spiffe/spire/test/plugintest"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+func TestV0GenerateKey(t *testing.T) {
+	for _, tt := range []struct {
+		test          string
+		err           error
+		publicKey     *keymanagerv0.PublicKey
+		expectCode    codes.Code
+		expectMessage string
+	}{
+		{
+			test:          "response missing key",
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): plugin response empty for key "foo"`,
+		},
+		{
+			test:          "response has mismatched key ID",
+			publicKey:     &keymanagerv0.PublicKey{Id: "bar"},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): plugin response has unexpected key id "bar" for key "foo"`,
+		},
+		{
+			test:          "response missing PKIX data",
+			publicKey:     &keymanagerv0.PublicKey{Id: "foo"},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): plugin response missing public key PKIX data for key "foo"`,
+		},
+		{
+			test:          "response has malformed PKIX data",
+			publicKey:     &keymanagerv0.PublicKey{Id: "foo", PkixData: []byte("malformed")},
+			expectCode:    codes.Internal,
+			expectMessage: `keymanager(test): unable to parse public key PKIX data for key "foo"`,
+		},
+		{
+			test:          "RPC fails",
+			err:           errors.New("ohno"),
+			expectCode:    codes.Unknown,
+			expectMessage: "keymanager(test): ohno",
+		},
+		{
+			test:       "success",
+			publicKey:  &keymanagerv0.PublicKey{Id: "foo", PkixData: testKeyPKIXData},
+			expectCode: codes.OK,
```

Comment:
is it relevant to verify status message when code is ok?

Reply:
It should verify as empty, but as you point out later, this doesn't actually happen for the Contains function. I will change that helper to assert an empty message when the code is OK.